readyapi4j-groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/pro/DataGenDataSourceTestStepDslTest.groovy

```diff
@@ -0,0 +1,186 @@
+        assert customStringDataGenerator.propertyName == 'ConstantString'
+        assert customStringDataGenerator.value == 'This is a custom string!'
+    }
+
+    @Test
+    void createsRecipeWithDataSourceTestStepWithShortStateNameDataGen() throws Exception {
+        TestRecipe testRecipe = recipe {
+            dataGenDataSource 'ShortStateNamesDataGenDataSource', {
+                abbreviatedStateNames 'StateNames'
+            }
+        }
+        DataGenDataSource dataSource = extractDataGenDataSource(testRecipe)
+        StateNameDataGenerator stateNameDataGenerator = dataSource.dataGenerators[0] as StateNameDataGenerator
+        assert stateNameDataGenerator.propertyName == 'StateNames'
+        assert stateNameDataGenerator.nameFormat == StateNameDataGenerator.NameFormatEnum.ABBREVIATED
+    }
+
+    @Test
+    void createsRecipeWithDataSourceTestStepWithFullStateNameDataGen() throws Exception {
+        TestRecipe testRecipe = recipe {
+            dataGenDataSource 'FullStateNamesDataGenDataSource', {
+                fullStateNames 'FullStateNames'
+            }
+        }
+        DataGenDataSource dataSource = extractDataGenDataSource(testRecipe)
+        StateNameDataGenerator stateNameDataGenerator = dataSource.dataGenerators[0] as StateNameDataGenerator
+        assert stateNameDataGenerator.propertyName == 'FullStateNames'
+        assert stateNameDataGenerator.nameFormat == StateNameDataGenerator.NameFormatEnum.FULL
+    }
+
+    @Test
+    void createsRecipeWithDataSourceTestStepWithDateTimeDataGen() throws Exception {
+        Date startDate = Date.parse("yyyy-MM-dd hh:mm:ss", "2014-04-03 1:23:45")
+        Date endDate = Date.parse("yyyy-MM-dd hh:mm:ss", "2016-04-03 23:59:59")
+
+        TestRecipe testRecipe = recipe {
+            dataGenDataSource 'FullStateNamesDataGenDataSource', {
+                dateAndTimeValues 'DateAndTimes', {
+                    startingAt startDate
+                    endingAt endDate
+                    formatISO_8601
+                }
+            }
+        }
+        DataGenDataSource dataSource = extractDataGenDataSource(testRecipe)
+        DateAndTimeDataGenerator dateAndTimeDataGenerator = dataSource.dataGenerators[0] as DateAndTimeDataGenerator
+        assert dateAndTimeDataGenerator.propertyName == 'DateAndTimes'
+        assert dateAndTimeDataGenerator.dateTimeFormat == YYYY_MM_DDTHH_MM_SSZ_ISO_8601_
+        assert dateAndTimeDataGenerator.minimumValue == startDate
+        assert dateAndTimeDataGenerator.maximumValue == endDate
+    }
+
+    private static DataGenDataSource extractDataGenDataSource(TestRecipe testRecipe) {
+        DataSourceTestStep testStep = testRecipe.testCase.testSteps[0] as DataSourceTestStep
+        testStep.dataSource.dataGen
+    }
+
+    private static void assertDataGenerator(DataGenerator dataGenerator, String type, String propertyName) {
```

Comment:
Since you're not saying what we assert, the test code becomes a bit unreadable.

verifyDataGeneratorValues, maybe?